src/game/Spells/SpellAuras.cpp

```diff
@@ -3253,12 +3253,50 @@ void Aura::HandleFeignDeath(bool apply, bool Real)
     if (!Real)
         return;
 
+    bool success = true;
     Unit* pTarget = GetTarget();
-    // Toutes les personnes qui castent sur le casteur de FD doivent etre interrompues.
+    
     if (apply)
-        pTarget->InterruptSpellsCastedOnMe();
+    {
+        HostileReference* pReference = pTarget->getHostileRefManager().getFirst();
+        while (pReference)
+        {
+            if (Unit* refTarget = pReference->getSourceUnit())
+            {
+                if (!refTarget->IsPlayer() && pTarget->MagicSpellHitResult(refTarget, GetHolder()->GetSpellProto(), nullptr) != SPELL_MISS_NONE)
+                {
+                    success = false;
+                    break;
+                }
+            }
+            pReference = pReference->next();
+        }
+
+        if (success)
+        {
+            // Interrupt everyone casting on feign caster
+            pTarget->InterruptSpellsCastedOnMe();
+        }
+        else
+        {
+            // Send error to client and remove aura if spell resisted
+            if (Player* plr = pTarget->ToPlayer())
+            {
+                plr->SendFeignDeathResisted();
+                plr->SendAttackSwingCancelAttack();
+            }
+
+            // prevent interrupt message
+            if (GetCasterGuid() == pTarget->GetObjectGuid())
+                pTarget->FinishSpell(CURRENT_GENERIC_SPELL, false);
+            pTarget->InterruptNonMeleeSpells(true);
+
+            GetHolder()->SetAuraDuration(0);
```

Comment:
I don't think you need to remove the aura or make it end - if it resists they should stay in FD but get wrecked by the mobs iirc. SetFeignDeath also handles calls like InterruptNonMeleeSpells as well. I think you should leave that logic in SetFeignDeath but add another "resist" param and selectively ignore certain functionality (such as setting death state, combatstop, etc)

Reply:
Alright changed it.